db-repo/CMakeLists.txt

```diff
@@ -19,11 +19,15 @@ set(BOOST_INCLUDE_DIR "/usr/local/include")
 set(BOOST_LIB_DIR "/usr/local/lib")
 
 find_package(Boost REQUIRED COMPONENTS program_options filesystem)
+include_directories("${EMERALDDB_PUBLIC_INCLUDE_DIR}")
 
 # create library based on bson source codes  
-add_library(bson "")
-target_sources(bson 
-  PRIVATE 
+add_library(emeralddb "")
```

Comment:
here we set the emeralddb as a library, then gonn create a file with the name of `edb`, finally after all functions implemented in emeralddb link the library to the executable file `edb`. and if we wanna add unit tests for emeralddb just create a binary file with the name of `edb_tests` and link emeralddb this binary to `edb_tests` to execute unit test case, and the same method can be used to `edb_benchmarks` to execute benchamrks methos.